src/idlcxx/src/types.c

```diff
@@ -45,16 +45,26 @@ emit_member(
   else
     type_spec = idl_type_spec(node);
 
+  const idl_member_t *mem = idl_parent(node);
+  assert(idl_is_member(mem));
+
   name = get_cpp11_name(node);
   if (IDL_PRINTA(&type, get_cpp11_type, type_spec, gen) < 0)
     return IDL_RETCODE_NO_MEMORY;
   type_spec = idl_strip(type_spec, IDL_STRIP_ALIASES | IDL_STRIP_FORWARD);
-  if (idl_is_array(type_spec))
+  value = NULL;
+  if (idl_is_array(type_spec)) {
     value = "{ }";
-  else if (!idl_is_enum(type_spec) && !idl_is_base_type(type_spec))
-    value = NULL;
-  else if (IDL_PRINTA(&value, get_cpp11_default_value, type_spec, gen) < 0)
-    return IDL_RETCODE_NO_MEMORY;
+  } else if (idl_is_enum(type_spec) || idl_is_base_type(type_spec) || idl_is_string(type_spec)) {
```

Comment:
... this is a little too convoluted for me!

Isn't it simply:
* if mem->value.annotation != nullptr (i.e. no default value annotation)
  * if is_base_type || is_enum
    * if print default value fails: return NO_MEMORY
  * (else nothing to be done)
* else
  * if is_base_type || is_string
    * if print custom default value fails: return NO_MEMORY
  * else return UNSUPPORTED

?

I am not actually sure where the bitmask would end up, so maybe the conflict is not so straightforward after all!